### DISCUSSION TITLE
Boolean Generation

### DISCUSSION THREAD

**Discussion by Fungusware** [fluentmigrator/fluentmigrator#1505] [Q&A]
I've noticed when using SQLite

.WithColumn("LicenseCreated").AsBoolean()

Generates an Integer field. It think in the newer versions of SQLite you can use a proper Boolean. This would make things better for using tools like Ling2DB and TT4 templates over the db.

Regards.

**Comment by fubar-coder** [comment]
IIRC, SQLite doesn't really have boolean columns. According to the [documentation](https://www.sqlite.org/datatype3.html), it only knows the following types:

- `NULL`
- `INTEGER`
- `REAL`
- `TEXT`
- `BLOB`

All other "types" are just aliases for the types above and are merely useful for documentation and reverse-engineering purposes, but it may cause wrong assumptions about the capabilities of the column.

The type `NUMERIC` seems to be a special case (see the paragraph about [Column Affinity](https://www.sqlite.org/datatype3.html#determination_of_column_affinity)).

**Comment by Fungusware** [comment]
That is indeed true, I just feel that since I can do 

CREATE TABLE Activation (
    Id                       INTEGER  NOT NULL PRIMARY KEY AUTOINCREMENT,
    UserKey                  TEXT     NOT NULL,
    ActivationDate           DATETIME NOT NULL,
    MachineAuthorizationCode TEXT     NOT NULL,
    IsDeactivation           BOOLEAN  NOT NULL
);

and then have LinqToDb autogenerate the POCO's it would be super cool to have FLuentMigrator be able to generate this also.

Either way, its a great tool, good work.

**Reply by fubar-coder** [reply]
Ah, I missed the point about the auto-generation of POCOs through LinqToDb. That sounds like a valid use case.

The current workaround would be to provide your own generator/column/type map. Example gist: https://gist.github.com/fubar-coder/1e8ba2c34ad95ae1cea3939b4e73e647

The `CustomSqliteGenerator`, `CustomSqliteColumn`, and `CustomSqliteTypeMap` are almost 1:1 copies from the FM source code. The only changes are:

- `CustomSqliteGenerator` uses `CustomSqliteColumn`
- `CustomSqliteColumn` users `CustomSqliteTypeMap`
- `CustomSqliteTypeMap` uses `BOOLEAN` instead of `INTEGER` for `DbType.Boolean`

EDIT: The sources might be outdated, as it was taken from an early 3.2 version, and the `AddCustomFluentMigratorSqliteGenerator.cs` file was written by hand (not really used or tested).

**Comment by Fungusware** [comment]
How would I go about using the 

```
                    .AddScoped<Data.CustomSqliteGenerator>()
                    .AddScoped<IMigrationGenerator>(sp => sp.GetRequiredService<Data.CustomSqliteGenerator>())
```

I tried in the initial create services section in the basic demo, the custome code never seems to be injected.

**Reply by fubar-coder** [reply]
That's strange. It should work if you add this service descriptions to your service collection - as it works for me.

EDIT: You should add the services **after** adding FluentMigrator.
